tools/cli/internal/cli/changelog/metadata/create.go

```diff
@@ -70,7 +71,10 @@ func (o *Opts) PreRun() error {
 		}
 	}
 
+	// Validate that the API version use the correct date format YYYY-MM-DD
 	for _, version := range o.versions {
+		// Upcoming version has the format YYYY-MM-DD.upcoming, here we remove .upcoming to validate the date format.
+		version = strings.ReplaceAll(version, ".upcoming", "")
```

Comment:
this is where I suggested to simply try to use `apiversion.go` and create a new version, throwing an error if it's invalid, that way you don't need to maintain two different logics to parse versions

Reply:
yes, good point. I updated the logic to reuse `apiversion.go`